spyder/api/shellconnect/main_widget.py

```diff
@@ -17,13 +24,34 @@
 from spyder.api.widgets.main_widget import PluginMainWidget
 from spyder.widgets.emptymessage import EmptyMessageWidget
 
+if TYPE_CHECKING:
+    from qtpy.QtWidgets import QWidget
+
+    import spyder.plugins.ipyconsole.widgets.shell
```

Comment:
The right import is `ipythonconsole`, not `ipyconsole`. Please fix it in all places you added it.